adapters/seedingAlliance/seedingAlliance.go

```diff
@@ -77,42 +77,41 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
 	bidResponse.Currency = response.Cur
 
+	var errs []error
+
 	for _, seatBid := range response.SeatBid {
 		for i := range seatBid.Bid {
 			resolvePriceMacro(&seatBid.Bid[i])
 
-			b := &adapters.TypedBid{
-				Bid:     &seatBid.Bid[i],
-				BidType: getMediaTypeForImp(seatBid.Bid[i].ImpID, request.Imp),
+			bidType, err := getMediaTypeForBid(seatBid.Bid[i].Ext)
+			if err != nil {
+				errs = append(errs, err)
+				continue
 			}
 
-			bidResponse.Bids = append(bidResponse.Bids, b)
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+			})
 		}
 	}
 
-	return bidResponse, nil
+	return bidResponse, errs
 }
 
 func resolvePriceMacro(bid *openrtb2.Bid) {
 	price := strconv.FormatFloat(bid.Price, 'f', -1, 64)
 	bid.AdM = strings.Replace(bid.AdM, "${AUCTION_PRICE}", price, -1)
 }
 
-func getMediaTypeForImp(impId string, imps []openrtb2.Imp) openrtb_ext.BidType {
-	var UnknownBidType openrtb_ext.BidType = "unknown"
+func getMediaTypeForBid(ext json.RawMessage) (openrtb_ext.BidType, error) {
+	var bidExt openrtb_ext.ExtBid
 
-	for _, imp := range imps {
-		if imp.ID == impId {
-			switch {
-			case imp.Native != nil:
-				return openrtb_ext.BidTypeNative
-			case imp.Banner != nil:
-				return openrtb_ext.BidTypeBanner
-			}
-		}
+	if err := json.Unmarshal(ext, &bidExt); err != nil || bidExt.Prebid == nil {
```

Comment:
Nitpick: `bidExt.Prebid == nil` should be a separate check. 
Please refer to this code for cleaner example: https://github.com/prebid/prebid-server/blob/f3e440389e3d57305be796c39359ff1fb09784b1/adapters/adf/adf.go#L143

Reply:
Don't think that example is any cleaner/more readable - actually way worse since the happy path is nested.
I separated the nil check though in the last commit.

Reply:
Ok, this works too!